src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -729,189 +729,194 @@ private String GetCodeForVirtualMethod(MethodDesc method, int slot)
             return sb.ToString();
         }
 
-        private void AppendVirtualSlots(CppGenerationBuffer sb, TypeDesc implType, TypeDesc declType)
+        private String GetCodeForNode(DependencyNode node, NodeFactory factory)
         {
-            var baseType = declType.BaseType;
-            if (baseType != null)
-                AppendVirtualSlots(sb, implType, baseType);
+            // virtual slots
+            var nodeData = (node as ObjectNode).GetData(factory, false);
 
-            IReadOnlyList<MethodDesc> virtualSlots = _compilation.NodeFactory.VTable(declType).Slots;
-            for (int i = 0; i < virtualSlots.Count; i++)
-            {
-                MethodDesc declMethod = virtualSlots[i];
-                MethodDesc implMethod = implType.GetClosestDefType().FindVirtualFunctionTargetMethodOnObjectType(declMethod);
+            CppGenerationBuffer nodeCode = new CppGenerationBuffer();
 
-                sb.AppendLine();
-                if (implMethod.IsAbstract)
-                {
-                    sb.Append("NULL,");
-                }
-                else
-                {
-                    sb.Append("(void*)&");
-                    sb.Append(GetCppMethodDeclarationName(implMethod.OwningType, GetCppMethodName(implMethod)));
-                    sb.Append(",");
-                }
-            }
-        }
+            /* Create list of byte data. Used to divide contents between reloc and byte data
+          * First val - isReloc
+          * Second val - size of byte data if first value of tuple is false
+          */
 
-        private String GetCodeForType(TypeDesc type)
-        {
-            var sb = new CppGenerationBuffer();
+            List<Tuple<bool, int>> nodeDataDivs = new List<Tuple<bool, int>>();
```

Comment:
Nit: Tuple<bool,int> is kind of hard to read - you have to keep in your head what Item1 and Item2 means. I think it would look better to define a helper struct with actual field names for it, like we do everywhere else (e.g. take `struct Relocation` as an example).